graylog2-web-interface/src/views/components/contexts/SearchPageLayoutProvider.tsx

```diff
@@ -14,46 +14,36 @@
  * along with this program. If not, see
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
+
 import * as React from 'react';
-import { useMemo } from 'react';
 
-import SearchPageLayoutContext from './SearchPageLayoutContext';
-import SearchPageLayoutState from './SearchPageLayoutState';
+import type { LayoutState } from 'views/components/contexts/SearchPageLayoutContext';
+import SearchPageLayoutContext, { FULL_MENU } from 'views/components/contexts/SearchPageLayoutContext';
 
-type Props = {
+type SearchPageLayoutProviderProps = {
   children: React.ReactNode
-  setLayoutState: (stateKey: string, value: boolean) => void,
-  getLayoutState: (stateKey: string, defaultValue: boolean) => boolean,
-};
+  providerOverrides?: LayoutState
+}
 
-const SearchPageLayoutProvider = ({ getLayoutState, setLayoutState, children }: Props) => {
-  const searchPageLayoutContextValue = useMemo(() => {
-    const config = {
-      sidebar: { isPinned: getLayoutState('sidebarIsPinned', false) },
-    };
-    const actions = { toggleSidebarPinning: () => setLayoutState('sidebarIsPinned', !config.sidebar.isPinned) };
+const defaultState = {
+  sidebar: { isShown: true },
+  viewActionsLayoutOptions: FULL_MENU,
+} as LayoutState;
 
-    return ({
-      config,
-      actions,
-    });
-  }, [getLayoutState, setLayoutState]);
+function SearchPageLayoutProvider({ children, providerOverrides = defaultState }: SearchPageLayoutProviderProps) {
```

Comment:
Do we actually need this default provider at all? From what I am seeing, we are passing the prop for the provider overrides from the `Search` component or use `defaultState` if that prop is not set. Instead, we could just use `defaultState` as the default value for the context (instead of `undefined`) and leave out the provider in the normal case. If we want to use the `Search` component with a different layout, we can just wrap it in a provider block like this:
```
<SearchPageLayoutContext.Provider value={...}>
  <Search />
</SearchPageLayoutContext.Provider>
```

Reply:
@dennisoelkers I went back forth on this, but ultimately I went with this because of the hook. I didn't like the possibility of a component consuming the hook outside the provider so I thought making its usage explicit was the way to go. Otherwise the hook would have to provide a fake piece of context if used outside the provider.

Reply:
What would be the problem with a component consuming the hook outside the provider when the context provides reasonable defaults?